$MirOS: ports/misc/mc/patches/patch-vfs_extfs_patchfs_in,v 1.1 2008/09/17 20:06:59 tg Exp $
--- vfs/extfs/patchfs.in.orig	Tue Nov 16 23:00:18 2004
+++ vfs/extfs/patchfs.in	Fri Dec 31 17:44:06 2010
@@ -12,6 +12,8 @@ use POSIX;
 use File::Temp 'tempfile';
 
 # standard binaries
+my $xz = 'xz';
+my $lzma = 'lzma';
 my $bzip = 'bzip2';
 my $gzip = 'gzip';
 my $fileutil = 'file';
@@ -70,7 +72,11 @@ sub myin
     my ($qfname)=(quotemeta $_[0]);
 
     $_=`$fileutil $qfname`;
-    if (/bzip/) {
+    if (/xz/) {
+	return "$xz -dc $qfname";
+    } elsif (/lzma/) {
+	return "$lzma -dc $qfname";
+    } elsif (/bzip/) {
 	return "$bzip -dc $qfname";
     } elsif (/gzip/) {
 	return "$gzip -dc $qfname";
@@ -86,7 +92,11 @@ sub myout
     my ($sep) = $append ? '>>' : '>';
 
     $_=`$fileutil $qfname`;
-    if (/bzip/) {
+    if (/xz/) {
+	return "$xz -c $sep $qfname";
+    } elsif (/lzma/) {
+	return "$lzma -c $sep $qfname";
+    } elsif (/bzip/) {
 	return "$bzip -c $sep $qfname";
     } elsif (/gzip/) {
 	return "$gzip -c $sep $qfname";
